source/Octopus.Server.Client/Operations/RegisterMachineOperation.cs

```diff
@@ -147,17 +154,34 @@ MachineResource GetMachine(IOctopusSpaceRepository repository)
         /// </exception>
         public override async Task ExecuteAsync(IOctopusSpaceAsyncRepository repository)
         {
-            var selectedEnvironments = GetEnvironments(repository).ConfigureAwait(false);
-            var machinePolicy = GetMachinePolicy(repository).ConfigureAwait(false);
-            var machineTask = GetMachine(repository).ConfigureAwait(false);
-            var tenants = GetTenants(repository).ConfigureAwait(false);
-            await ValidateTenantTags(repository).ConfigureAwait(false);
-            var proxy = GetProxy(repository).ConfigureAwait(false);
+            var machine = await GetMachine(repository).ConfigureAwait(false);
+            var proxy = await GetProxy(repository).ConfigureAwait(false);
 
-            var machine = await machineTask;
-            ApplyBaseChanges(machine, await machinePolicy, await proxy);
-            ApplyDeploymentTargetChanges(machine, await selectedEnvironments, await tenants);
+            if (machine.Id == null || AllowOverwrite)
```

Comment:
a null `Id` here represents a new machine resource correct?

Reply:
Yep correct. I've updated the code to make that a bit more obvious 👍